File: typescript/rna-transcription/rna-transcription.ts
const dict: {[k:string]:string} = { G: "C", C: "G", T: "A", A: "U"}

const getThrow = (c: string) => {
    const v = dict[c];
    if (!v) throw Error('Invalid input DNA.')
    return v;
}

export const toRna = (s: string) => s.split("").map(getThrow).join("")

File: julia/rna-transcription/rna-transcription.jl
function to_rna(dna::AbstractString)
 dict = Dict('A' => 'U', 'C' => 'G', 'G' => 'C', 'T' => 'A');
 try
   map(nucleotide->dict[nucleotide], dna)
 catch error
   throw(ErrorException("Invalid nucleotide"));
 end
end
